connaisseur/validators/cosign/cosign_validator.py

```diff
@@ -45,8 +47,18 @@ async def validate(
 
         self.vals = self.__get_pinned_keys(trust_root, required)
 
-        tasks = [self.__validation_task(k, image) for k in self.vals.keys()]
-        await asyncio.gather(*tasks)
+        # use concurrent.futures for now
+        # tasks = [self.__validation_task(k, image) for k in self.vals.keys()]
+        # await asyncio.gather(*tasks)
+
+        # prepare executor
+        num_workers = len(self.vals)
+        executor = ThreadPoolExecutor(num_workers)
+        # prepare tasks
+        arguments = [(k, image) for k in self.vals.keys()]
+        futures = [executor.submit(self.__validation_task, *arg) for arg in arguments]
+        # await results (output dropped as `self.vals` is updated within function)
+        _ = [future.result() for future in futures]
```

Comment:
Why do an assignment?

Seems like you intend to do:
```python
for future in futures:
    future.result()
```

Reply:
I had previously constructed a single-line solution and pylint was complaining about unassigned outcome